gdal/swig/python/gdal-utils/osgeo_utils/samples/gdallocationinfo.py

```diff
@@ -0,0 +1,383 @@
+
+    parser.add_argument("-xml", dest="xml", action="store_true",
+                        help="The output report will be XML formatted for convenient post processing.")
+
+    parser.add_argument("-lifonly", dest="lifonly", action="store_true",
+                        help="The only output is filenames production from the LocationInfo request against "
+                             "the database (i.e. for identifying impacted file from VRT).")
+
+    parser.add_argument("-valonly", dest="valonly", action="store_true",
+                        help="The only output is the pixel values of the selected pixel on each of the selected bands.")
+
+    parser.add_argument("-plb", dest="plb", action="store_true",
+                        help="The output will be a series of lines in the form of Pixel,Line,band0,band1... "
+                             "for each of the selected bands.")
+
+    parser.add_argument("-quiet", dest="quiet", action="store_true",
+                        help="No output will be printed.")
+
+    parser.add_argument("-extent_strict", dest="allow_xy_outside_extent", action="store_false",
+                        help="If set, input points outside the raster extent will raise an exception, "
+                             "otherwise a warning will be issued (unless quiet).")
+
+    parser.add_argument("-interp", dest="resample_alg", action="store_true",
+                        help="If set, a Bilinear interpolation would be used, otherwise the NearestNeighbour sampling.")
+
+    parser.add_argument("-b", dest="band_nums", metavar="band", type=int, nargs='+',
+                        help="Selects a band to query. Multiple bands can be listed. By default all bands are queried.")
+
+    parser.add_argument('-overview', dest="ovr_idx", metavar="overview_level", type=int,
+                        help="Query the (overview_level)th overview (overview_level=1 is the 1st overview), "
+                             "instead of the base band. Note that the x,y location (if the coordinate system is "
+                             "pixel/line) must still be given with respect to the base band.")
+
+    parser.add_argument("-l_srs", dest="srs", metavar='srs_def', type=str,
+                        help="The coordinate system of the input x, y location.")
+
+    parser.add_argument("-geoloc", dest="geoloc", action="store_true",
+                        help="Indicates input x,y points are in the georeferencing system of the image.")
+    parser.add_argument("-llgeoloc", dest="llgeoloc", action="store_true",
+                        help="Indicates input x,y points are in the long, lat (geographic) "
+                             "based on the georeferencing system of the image.")
+    parser.add_argument("-wgs84", dest="wgs84", action="store_true",
+                        help="Indicates input x,y points are WGS84 long, lat.")
+    parser.add_argument("-epsg", dest="epsg", metavar='epsg', type=int,
```

Comment:
Not really sure we need this option. The list of options becomes quite confusing. -l_srs EPSG:XXXX should do the job, and it is how other utilities handle such case. And probably -l_srs WGS84 would also do the job

Reply:
OK, removing -epsg. I'm leaving -wgs84 to be compatible with the cpp util.